packages/playwright-core/src/utils/utils.ts

```diff
@@ -405,6 +405,7 @@ export function arrayToObject(array?: NameValue[]): { [key: string]: string } |
 export async function calculateFileSha1(filename: string): Promise<string> {
   const hashStream = new HashStream();
   const stream = fs.createReadStream(filename);
+  stream.once('error', err => hashStream.emit('error', err));
```

Comment:
Looks like this is needed as error streams don't always forward their errors. Although, something smells since this case looks eerily similar to https://github.com/microsoft/playwright/blob/359d523ec318106bf79d650b2b4458a1a388ebdf/tests/playwright-test/reporter-attachment.spec.ts#L89

I don't know all the intricicies of Node.js streams, so HashStream implementation should also be looked at.

Reply:
Reading up more on streams, this looks to be the correct fix for `calculateFileSha1 ` so we don't have an unhandled error.

Before this change, we were triggering https://github.com/microsoft/playwright/blob/da1f39fb29dd3c45a8f26de77f029c2bfc09a790/packages/playwright-test/src/workerRunner.ts#L94 since we weren't forwarding the error and it was surfacing as an unhandled error in both the new test fixture cases and the old (but seemingly "working")  https://github.com/microsoft/playwright/blob/359d523ec318106bf79d650b2b4458a1a388ebdf/tests/playwright-test/reporter-attachment.spec.ts#L89. 

The discrepancy in behavior likely has to do with how unhandled errors are depending on where/when they come: https://github.com/microsoft/playwright/blob/da1f39fb29dd3c45a8f26de77f029c2bfc09a790/packages/playwright-test/src/workerRunner.ts#L94

So, an additional fix is likely needed since I _think_ we'd want the fixture case to also immediately fail and not hang even if there is an unhandled error.

Reply:
> So, an additional fix is likely needed since I think we'd want the fixture case to also immediately fail and not hang even if there is an unhandled error.

`calculateFileSha1` will no longer be the culprit with this fix, but other code could intro unhandled errors in the future.

Reply:
cc: @dgozman since I think you'll be able to explain to me the expectations around how unhandled errors in fixtures vs. tests should behave overall.